File: src/me/ch2/jisho.py
import sys

dicfile = "./resources/ejdict-hand-utf8.txt"
if len(sys.argv) < 2:
    print("Usage: {} <word>".format(sys.argv[0]))
    quit()

word = sys.argv[1]
with open(dicfile, 'rt') as f:
    while True:
        line = f.readline()
        if not line:
            break
        if word in line:
            print(line.strip())

File: src/me/ch2/jisho.rs
use std::fs::File;
use std::io::{BufRead, BufReader};

fn main() {
    let dicfile = "./resources/ejdict-hand-utf8.txt";
    let args: Vec<String> = std::env::args().collect();
    if args.len() < 2 {
        println!("Usage: {} <word>", args[0]);
        return;
    }

    let word = &args[1];
    let fp = File::open(dicfile).unwrap();
    let reader = BufReader::new(fp);
    for line in reader.lines() {
        let line = line.unwrap();
        if line.find(word) == None {
            continue;
        }
        println!("{}", line);
    }
}
